src/openeo_processes/arrays.py

```diff
@@ -1668,3 +1676,179 @@ def exec_xar(data, asc=True, nodata=None, dimension=None):
+            pts_each_polygon = np.floor(pts_each_polygon).astype(int) # or np.round
+            pts_each_multipolygon = np.floor(pts_each_multipolygon).astype(int) # or np.round
+
+            # Check if there are polygons which will have a single point
+            pts_each_polygon[pts_each_polygon==0] = 1
+            pts_each_multipolygon[pts_each_multipolygon==0] = 1
+
+        elif geometry_count is not None and total_count is None:
+            pts_each_polygon = np.ones(len(polygons_gdf['geometry']),dtype=int) * geometry_count
+            pts_each_multipolygon = np.ones(len(multipolygons_gdf['geometry']),dtype=int) * geometry_count   
+
+        elif geometry_count is not None and total_count is not None:
+            if total_count < features_count:
+                raise Exception("CountMismatch - The maximum number of points must be >= the number of features of the provided geoJSON!\
+             total_count = {} numb of features = {}".format(total_count,features_count))
+            else:
+                # See https://github.com/Open-EO/openeo-processes/pull/315#discussion_r821457646
+                # TODO
+                raise Exception("Not yet implemented. We need to decide how to handle this.")
+
+        # Loop over all the Polygons:
+        for idx, polygon in polygons_gdf.iterrows():
+            # find area bounds
+            bounds = polygon.geometry.bounds
+            xmin, ymin, xmax, ymax = bounds
+
+            xext = xmax - xmin
+            yext = ymax - ymin
+
+            count = 0
+            points_list = []
+            for i in range(pts_each_polygon[idx]):
+                # generate a random x and y
+                x = xmin + random.random() * xext
+                y = ymin + random.random() * yext
+                p = Point(x, y)
+                if polygon.geometry.contains(p):  # check if point is inside geometry
+                    points_list.append(p)
+                    count += 1
+
+            # Return te centroid if we were not able to find random points within the polygon
+            if count == 0:
+                p = polygon.geometry.centroid
+                if polygon.geometry.contains(p):  # check if point is inside geometry
+                    points_list.append(p)
+                else:
+                    pass
+                    # TODO: Edge case. Return empty feature or first coordinate of the geometry?
+
+            multipoint = polygon.copy()
+            if len(points_list)>1:
+                multipoint.geometry = MultiPoint(points_list)
+                points_gdf = points_gdf.append(multipoint,ignore_index=True)
+
+            elif len(points_list)==1:
+                multipoint.geometry = points_list[0]
+                points_gdf = points_gdf.append(multipoint,ignore_index=True)
+
+
+        # Loop over all the MultiPolygons:
+        # polygons_from_multipolygons_gdf = multipolygons_gdf.explode()
+        for idx, polygon in multipolygons_gdf.iterrows():
```

Comment:
In the current form the logic in this loop is exactly the same as the loop for Polygons above - @clausmichele are you aware of any edgecase handling that wants to be different for Polygons/Multipolygons? Otherwise I'd suggest folding this into one loop to save some lines of code!

Reply:
Ah, yes, I understand that, I meant that specifically the point-generation loops for the Polygons (ln1780-1815) and then the MultiPolygons (ln1818-1856) seem to be doing the exact same thing, so couldn't we do something like this?

```suggestion
        for idx, polygon in pd.concat(polygons_gdf, multipolygons_gdf).iterrows():
```

More of a nitpick though!